apps/meteor/client/views/admin/customSounds/EditCustomSound.tsx

```diff
@@ -15,9 +15,14 @@ type EditCustomSoundProps = {
 function EditCustomSound({ _id, onChange, ...props }: EditCustomSoundProps): ReactElement {
 	const query = useMemo(() => ({ query: JSON.stringify({ _id }) }), [_id]);
 
-	const { value: data, phase: state, error, reload } = useEndpointData('/v1/custom-sounds.list', { params: query });
+	const getSounds = useEndpoint('GET', '/v1/custom-sounds.list');
 
-	if (state === AsyncStatePhase.LOADING) {
+	const { data, isLoading, error, refetch } = useQuery(['custom-sounds', query], async () => {
+		const sound = await getSounds(query);
+		return sound;
+	});
```

Comment:
To avoid a assertion on `error`, let's improve this query a little bit:

```suggestion
	const dispatchToastMessage = useToastMessageDispatch();

	const { data, isLoading, refetch } = useQuery(
		['custom-sounds', _id],
		async () => {
			const { sounds } = await getSounds({ query: JSON.stringify({ _id }) });

			if (sounds.length === 0) {
				throw new Error('Custom sound not found');
			}

			return sounds[0];
		},
		{
			onError: (error) => {
				dispatchToastMessage({ type: 'error', message: error });
			},
		},
	);
```

Doing it like that, we gain:

1. Data cached as a single element instead of a array of one element;
2. Different handling of loading and errored states.

Reply:
Thank you! This way of handling errors looks much better.
I have a question, the call to `getSounds` returns:
```
count: number;
    offset: number;
    total: number;
    sounds: {
        name: string;
        statusType: string;
        _id: string;
        _updatedAt: string;
    }[];
```
So can we use the `total` value in place of `sounds.length`?

Reply:
And could we then replace this block 
```
if (error || !data || data.sounds.length < 1) {
		return (
			<Box fontScale='h2' pb='x20'>
				{(error as Error).message}
			</Box>
		);
	}
```
To:
```
if (!data) {
		return <></>
	}
```

Reply:
> So can we use the `total` value in place of `sounds.length`?

Not at all. I think `total` in this payload is related to pagination, e.g., there is 50 (`total`) custom sounds but you can fetch 10 (`sounds.length`) per page. We are only interested in getting one and, if there is something wrong, it can be zero.

 

> `return <></>`

I recommend returning `null`. It's the simplest acceptable return type of a React component. `<></>` is actually `<Fragment></Fragment>`, which becomes `createElement(Fragment)` after transpiling the JSX to JS. It's a waste of resources.

Reply:
Oh, so it was a different `total` thank you for the heads up. 
> I recommend returning null. It's the simplest acceptable return type of a React component. <></> is actually <Fragment></Fragment>, which becomes createElement(Fragment) after transpiling the JSX to JS. It's a waste of resources.

Uh, I tried to return `null`. However, the Type errors suddenly popped up. It was
```
Type 'null' is not assignable to type 'ReactElement<any, string | JSXElementConstructor<any>>'
```
So I planned to return `<></>` instead of null. Or is there a workaround for this? Could you please point me toward it?

Thank you!

Reply:
Change the function return type from `ReactElement` to `ReactElement | null`; it's safe to do it.

Reply:
Thank you!